pages/signup.tsx

```diff
@@ -0,0 +1,294 @@
+                isChecked={termsAgreement.saveEmail}
+                isRequired={true}
+                onChange={() => {
+                    setTermsAgreement({
+                        ...termsAgreement,
+                        saveEmail: !termsAgreement.saveEmail,
+                    });
+                    validateTermsAgreement({
+                        ...termsAgreement,
+                        saveEmail: !termsAgreement.saveEmail,
+                    });
+                }}
+            />
+            <CheckboxComp
+                mb={5}
+                text="I agree to have my Tool Progress saved and tracked"
+                isChecked={termsAgreement.saveProgress}
+                isRequired={true}
+                onChange={() => {
+                    setTermsAgreement({
+                        ...termsAgreement,
+                        saveProgress: !termsAgreement.saveProgress,
+                    });
+                    validateTermsAgreement({
+                        ...termsAgreement,
+                        saveProgress: !termsAgreement.saveProgress,
+                    });
+                }}
+            />
+            <CheckboxComp
+                text="I agree to to participate in a brief survey and share my demographic data with HUG "
+                isChecked={termsAgreement.participateSurvey}
+                isRequired={false}
+                onChange={() => {
+                    setTermsAgreement({
+                        ...termsAgreement,
+                        participateSurvey: !termsAgreement.participateSurvey,
+                    });
+                    validateTermsAgreement({
+                        ...termsAgreement,
+                        participateSurvey: !termsAgreement.participateSurvey,
+                    });
+                }}
+            />
+        </Box>
+    );
+
+    const stages = [
+        {
+            title: "Create an Account",
+            component: signupEmailStage,
+            buttonText: "Continue",
+            validation: validateEmail,
+        },
+        {
+            title: "Create an Account",
+            component: signupNameStage,
+            buttonText: "Continue",
+            validation: validateName,
+        },
+        {
+            title: "Create an Account",
+            component: signupPasswordStage,
+            buttonText: "Continue",
+            validation: validatePassword,
+        },
+        {
+            title: "Terms Agreement",
+            component: termsAgreementStage,
+            buttonText: "Sign up",
+            validation: validateTermsAgreement,
+        },
+        {
+            component: (
+                <Box mt={24}>
+                    <Icon as={IoCheckmarkCircle} boxSize={24}></Icon>
+                    <Heading>You're all set</Heading>
+                </Box>
+            ),
+            buttonText: "Get started",
+        },
+    ];
+
+    const incrementStage = () => {
```

Comment:
Since you're checking for specific indices here, might as well compare the text? As in for stage before signup you can have something like `stages[stages.length].title === "Create an Account"`. Just wanted to see if that might be cleaner